allennlp/data/tokenizers/pretrained_transformer_tokenizer.py

```diff
@@ -230,12 +221,15 @@ def tokenize(self, text: str) -> List[Token]:
         """
         This method only handles a single sentence (or sequence) of text.
         """
+        max_length = self._max_length
+        if max_length is not None and self._add_special_tokens:
+            max_length -= self.num_special_tokens_for_sequence()
+
         encoded_tokens = self.tokenizer.encode_plus(
             text=text,
             add_special_tokens=False,
-            max_length=self._max_length,
+            max_length=max_length,
             stride=self._stride,
-            truncation=self._truncation_strategy if self._max_length is not None else False,
```

Comment:
Are you sure this doesn't work anymore? It should: https://github.com/huggingface/transformers/blob/3323146e904a1092def4e8527de9d2a7479c1c14/src/transformers/tokenization_utils_base.py#L1990

Reply:
It only makes a difference when you pass in more than one string, which we never do.

Reply:
Oh I see